Frontend_Web_Development_React_Angular_Vue/React/Week 7/State Management in React.md

```diff
@@ -0,0 +1,273 @@
+<h1 align="center">  State Management in React  </h1>
+
+## :arrow_right: What is State Management in React  ?
+
+State management describes the maintenance of the state/knowledge of an application depending on all inputs. Inputs usually take place on the server (API) or the client (user).
+The state describes the condition of an application at a given point of time.The state of any application is represented by the user interface of the application. The state is mutable and whenever the user interacts with the application and changes its state, the user interface of the app may change as it will be represented by the new state.
+Sub-types of the state are:
+* Server state
+* Navigation state
+* Local UI state
+* Global UI state
+
+Accordingly, the difficulty of state management arises from the coordination of all subtypes of state in an application.
+Decision-making process for choosing the right state management solution is shown below:-
+<p align="center"><img src="https://user-images.githubusercontent.com/72224843/141657656-8f1f9a2d-94e5-4995-a80e-3acabb64f519.png" align="center" ></p>
+
+
+## :point_right: useState
+
+First, is our state shared by more than two components? If not, then the next question is whether our state has complex update logic. In other words, is the new state dependent on the old state or are multiple subvalues changed? If the answer to these questions is no, we use local state with useState.
+For example, if we have a simple counter with one component, the use of local state with useState is sufficient.
+
+* App.js
+```
+import React, { useState } from 'react';
+
+function App() {
+  const [click, setClick] = useState(0);
+  // using array destructuring here
+  // to assign initial value 0
+  // to click and a reference to the function
+  // that updates click to setClick
+  return (
+    <div>
+       
+ 
+<p>You clicked {click} times</p>
+ 
+ 
+      <button onClick={() => setClick(click + 1)}>
+        Click me
+      </button>
+    </div>
+  );
+}
+ 
+export default App;
+```
```

Comment:
make this code snippets of the jsx
just in starting of the code after the 3 back-ticks add jsx (```jsx) just like this.